### PR TITLE
User controller & db interface

### LINKED ISSUE #49: BE Create User Controller Class
Class which accesses and manages the User table in the DB

GET /api/user/getAllUsers
Get all users

GET /api/user/getUserById/{id}
Get user information by id

POST /api/user/PostUser
Creates a new user

PATCH /api/user/PatchUserById/{id}
Edit a current user by id

DELETE /api/user/deleteUserById/{id}
Delete a user with a given Id

Make unit-tests for each API function

### PR DESCRIPTION
**Pull request by cainsusk**
Adds:

- A DB interface for performing SQL queries
- A user controller, which -- using the DB interface -- can add users as well as sign in users (using a hashed password)

closes #49

### CODE REVIEW

File: app/app.py [hunk 1]
```diff
@@ -1,11 +1,15 @@
...
+from pbkdf2 import verify_password
+import users
 
 app = Flask(__name__)
 
+
 @app.route('/')
-def display_data():    
+def display_data():
+    return "hello world"
```
Comment (manda-02) [inline app/app.py:12]:
```suggestion
```

File: app/users.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+
+def put(filters: dict, user_update: dict):
+    # execute_sql_query("UPDATE", "Users",
+    #                   data=[user_update], conditions=filters)
+    return "{}, {}".format([user_update], filters)
+
+# cols = tuple(field for field in users[0])
+# data = [tuple(u[f] for f in u) for u in users]
+# return db.insert_into_table("Users", cols, data)
```
Comment (manda-02) [inline app/users.py:59]: delete if not being used??

**Review by manda-02 — COMMENTED** [review]
Overall, looks awesome - just left a few very minor comments 👍

**Review by garyfarb — APPROVED** [review]
Looks good to me as well.

**Review by victortarn — APPROVED** [review]
Looks good, the put user name functionality need s to be updated

File: app/users.py (hunk 2 above)
Comment (cainsusk) [reply app/users.py]: Yup I got rid of these !

File: app/app.py [hunk 3]
```diff
@@ -15,5 +19,48 @@ def display_data():
...
+    id, hash = db.execute_sql_query("SELECT", "Users", conditions={
+                                    "email": email}, columns=["id", "hash"])[0]
+    if verify_password(hash, pword):
+        return id
+    return "error"
+
+
+@app.route('/users/put/name/<id>/<name>')
+def put_user_update(id, name):
+    return users.put(dict({"id": "603619278311385986"}), dict({"username": "ki"}))
```
[the remark this answers is not in the document]
Comment (cainsusk) [reply app/app.py]: Yea Im trying to fix this now -- I think Im getting somewhere

**Comment by cainsusk** [comment]
now that the db interface is fully working, this closes #62

**Pull request merged**